lib/github_changelog_generator/reader.rb

```diff
@@ -0,0 +1,74 @@
+#
+# Author:: Enrico Stahn <mail@enricostahn.com>
+#
+# Copyright 2014, Zanui, <engineering@zanui.com.au>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module GitHubChangelogGenerator
+  # A Reader to read an existing ChangeLog file and return a structured object
+  #
+  # Example:
+  #   reader = GitHubChangelogGenerator::Reader.new
+  #   content = reader.read('./CHANGELOG.md')
+  class Reader
+    # Parse a single heading and return a Hash
+    #
+    # The following heading structures are currently valid:
+    # - ## [v1.0.2](https://github.com/zanui/chef-thumbor/tree/v1.0.1) (2015-03-24)
+    # - ## [v1.0.2](https://github.com/zanui/chef-thumbor/tree/v1.0.1)
+    # - ## v1.0.2 (2015-03-24)
+    # - ## v1.0.2
+    #
+    # @param [String] heading Heading from the ChangeLog File
+    # @return [Hash] Returns a structured Hash with version, url and date
+    def parse_heading(heading)
+      structures = [
+          /^## \[(?<version>v.+?)\]\((?<url>.+?)\)( \((?<date>.+?)\))?$/,
+          /^## (?<version>v.+?)( \((?<date>.+?)\))?$/,
```

Comment:
@estahn  Most of all project uses prefix "v" for version tags, but not all of them.
As example of big projects, that don't use:
- https://github.com/jquery/jquery
- https://github.com/bartaz/impress.js
- https://github.com/jashkenas/backbone

and this project also don't use any prefix for the tag versions.

So, I think it's needed to make char "v" optional in this regex. What do you think?